ddtrace/internal/ci_visibility/api/_test.py

```diff
@@ -128,29 +153,122 @@ def finish_itr_skipped(self) -> None:
         self.mark_itr_skipped()
         self.finish_test(TestStatus.SKIP)
 
-    def make_early_flake_retry_from_test(self, original_test_id: InternalTestId, retry_number: int) -> None:
-        if self.parent is None:
-            raise ValueError("Cannot make early flake retry from test without a parent")
-
-        new_test_id = InternalTestId(
-            original_test_id.parent_id, original_test_id.name, original_test_id.parameters, retry_number
-        )
-        self.parent.add_child(
-            new_test_id,
-            self.__class__(
-                original_test_id.name,
-                self._session_settings,
-                parameters=self._parameters,
-                codeowners=self._codeowners,
-                source_file_info=self._source_file_info,
-                initial_tags=self._tags,
-                is_early_flake_retry=True,
-            ),
+    def make_early_flake_retry_from_test(self, retry_number: int) -> "TestVisibilityTest":
+        if self._parameters is not None:
+            raise ValueError("Cannot create an early flake retry from a test with parameters")
+        retry_test = self.__class__(
+            self.name,
+            self._session_settings,
```

Comment:
Is `parameters` missing intentionally?

I wonder if we can do better than just remembering to keep the field list in sync with this call.